test/camel_patrol/middleware_test.rb

```diff
@@ -42,9 +42,21 @@
         env_params.merge('CONTENT_TYPE' => 'application/json; charset=utf-8')
       end
 
-      it 'snake_cases the params' do
-        assert_nil incoming[:params]['dwelling']['squareFootage']
-        refute_nil incoming[:params]['dwelling']['square_footage']
+      describe "when rack input not empty" do
+        let(:rack_input_json) { params.to_json }
+
+        it 'snake_cases the params' do
+          assert_nil incoming[:params]['dwelling']['squareFootage']
+          refute_nil incoming[:params]['dwelling']['square_footage']
+        end
+      end
+
+      describe "when rack input empty" do
+        let(:rack_input_json) { '' }
```

Comment:
This test without non-test changes, re-creates the 
`JSON::ParserError: A JSON text must at least contain two octets!` 
error. 

to recreate the error in an irb console:
`JSON.parse(StringIO.new('').read)`